### PR TITLE
Stream bindings: Out-of-band block size record

### PR DESCRIPTION
**Pull request by tSed**

### CODE REVIEW

**Comment by tSed** [comment]
This PR implements what was discussed on https://github.com/python-lz4/python-lz4/pull/179#issuecomment-549820488

**Comment by bbayles** [comment]
This works perfectly for me - great work!

File: lz4/stream/_stream.c [hunk 1]
```diff
@@ -888,13 +889,19 @@ _create_context (PyObject * Py_UNUSED (self), PyObject * args, PyObject * kwds)
...
+   *
+   * In out-of-band block size case, use a best-effort strategy for scaling
+   * buffers.
+   */
   if (context->config.direction == COMPRESS)
     {
-      uint32_t store_max_size = _GET_MAX_UINT32(store_comp_size);
+      uint32_t store_max_size = _GET_MAX_UINT32((store_comp_size == 0) ?
+                                                4 :
+                                                store_comp_size);
```
Comment (jonathanunderwood) [inline lz4/stream/_stream.c:901]: This indentation is very weird and hard to read. Can you improve the readability of this ternary operator? I'd even go as far as saying I'd prefer an explicit if/else than this.

File: lz4/stream/_stream.c [hunk 2]
```diff
@@ -919,7 +926,10 @@ _create_context (PyObject * Py_UNUSED (self), PyObject * args, PyObject * kwds)
     }
   else /* context->config.direction == DECOMPRESS */
     {
-      uint32_t store_max_size = _GET_MAX_UINT32(store_comp_size);
+      uint32_t store_max_size = _GET_MAX_UINT32((store_comp_size == 0) ?
+                                                4 :
+                                                store_comp_size);
+
```
Comment (jonathanunderwood) [inline lz4/stream/_stream.c:932]: As above.

File: lz4/stream/_stream.c [hunk 3]
```diff
@@ -1318,7 +1336,11 @@ _decompress (PyObject * Py_UNUSED (self), PyObject * args)
       goto exit_now;
     }
 
-  if (source.len > _GET_MAX_UINT32(context->config.store_comp_size))
+  source_size_max = _GET_MAX_UINT32((context->config.store_comp_size == 0) ?
+                                    4 :
+                                    context->config.store_comp_size);
+
```
Comment (jonathanunderwood) [inline lz4/stream/_stream.c:1342]: As above.

File: lz4/stream/_stream.c (hunk 1 above)
Comment (tSed) [reply lz4/stream/_stream.c]: done

**Comment by jonathanunderwood** [comment]
Thanks for your work on this, much appreciated.

**Pull request merged**